File: src/main/java/com/github/jactor/persistence/repository/UserRepository.java
package com.github.jactor.persistence.repository;

import com.github.jactor.persistence.entity.UserEntity;
import com.github.jactor.persistence.entity.UserEntity.UserType;
import java.util.Collection;
import java.util.List;
import java.util.Optional;
import org.springframework.data.repository.CrudRepository;

public interface UserRepository extends CrudRepository<UserEntity, Long> {

  Optional<UserEntity> findByUsername(String username);

  List<UserEntity> findByUserTypeIn(Collection<UserType> userType);
}

File: src/main/kotlin/com/github/jactor/persistence/repository/UserRepository.kt
package com.github.jactor.persistence.repository

import com.github.jactor.persistence.entity.UserEntity
import org.springframework.data.repository.CrudRepository
import java.util.Optional
import java.util.UUID

interface UserRepository : CrudRepository<UserEntity, UUID> {
    fun findByUsername(username: String): Optional<UserEntity>
    fun findByUserTypeIn(userType: Collection<UserEntity.UserType>): List<UserEntity>
}
